docs/API/V1/contents.md

```diff
@@ -0,0 +1,142 @@
+# `GET /api/v1/datasets/contents/<dataset><path>`
+
+This API returns an `application/json` document describing a file or the
+content of a directory at a specified `<path>` within the `<dataset>` tarball
+representation.
+
+## URI parameters
+
+`<dataset>` string \
+The resource ID of a dataset on the Pbench Server.
+
+`<path>`    string \
+The path of an item in the dataset inventory, as captured by the Pbench Agent
+packaging; for example, `/` for the root, or `/1-default/` for the default
+first iteration directory.
+
+## Request headers
+
+`authorization: bearer` token \
+*Bearer* schema authorization is required to access any non-public dataset.
+E.g., `authorization: bearer <token>`
+
+## Response headers
+
+`content-type: application/json` \
+The return is a serialized JSON object with information about the named file.
+
+## Response status
+
+`401`   **UNAUTHORIZED** \
+The client did not provide an authentication token and there is no public
+dataset with the resource ID `<dataset>`.
+
+`403`   **FORBIDDEN** \
+The named `<dataset>` is not public and the authenticated user lacks
+authorization to read it.
+
+`404`   **NOT FOUND** \
+Either the `<dataset>` or the relative `<path>` within the dataset does not
+exist.
+
+## Response body
+
+This API returns an `application/json` response body consisting of a JSON
+object describing the file or directory at a target `<path>` within a dataset
+tarball.
+
+When the `<path>` refers to a directory, the response object is described in
+[Directory object](#directory-object); when the `<path>` refers to a file, the
+response object is described in [File object](#file-object).
+
+### Directory object
+
+When the `<path>` refers to a directory within the dataset representation,
+Pbench returns a JSON object with two list fields:
+* `directories` is a list of [subdirectory objects](#subirectory-object), and
+* `files` is a list of [file objects](#file-object)
+
+```json
+{
+    "directories": [
+        {
+            "name": "1-iter1",
+            "uri": "http://hostname/api/v1/datasets/contents/<id><path>/1-iter1"
+        },
+        {
+            "sysinfo",
+            "uri": "http://hostname/api/v1/datasets/contents/<id><path>/sysinfo"
+        }
+        [...]
```

Comment:
Line 69 should end in a comma.

As for line 70, I would keep the elipsis but omit the square brackets (since in this context they would indicate a list).  Ditto line 89, and elsewhere.